lib/core/proxy.c

```diff
@@ -198,9 +198,9 @@ static void build_request(h2o_req_t *req, h2o_iovec_t *method, h2o_url_t *url, h
 
     /* headers */
     h2o_iovec_vector_t cookie_values = {NULL};
-    {
+    int found_early_data = 0;
+    if (H2O_LIKELY(req->headers.size != 0)) {
         const h2o_header_t *h, *h_end;
-        int found_early_data = 0;
         for (h = req->headers.entries, h_end = h + req->headers.size; h != h_end; ++h) {
```

Comment:
Can we do `for (const h2o_header_t *h = req->headers.entries, *h_end = h + req->headers.size; ...` here as well?